code/datums/beam.dm

```diff
@@ -118,34 +166,110 @@
 			Pixel_y = round(cos(Angle)+32*cos(Angle)*(N+16)/32)
 
 		//Position the effect so the beam is one continous line
-		var/a
+		var/final_x = segment.x
+		var/final_y = segment.y
 		if(abs(Pixel_x)>32)
-			a = Pixel_x > 0 ? round(Pixel_x/32) : CEILING(Pixel_x/32, 1)
-			X.x += a
+			final_x += Pixel_x > 0 ? round(Pixel_x/32) : CEILING(Pixel_x/32, 1)
 			Pixel_x %= 32
 		if(abs(Pixel_y)>32)
-			a = Pixel_y > 0 ? round(Pixel_y/32) : CEILING(Pixel_y/32, 1)
-			X.y += a
+			final_y += Pixel_y > 0 ? round(Pixel_y/32) : CEILING(Pixel_y/32, 1)
 			Pixel_y %= 32
 
-		X.pixel_x = Pixel_x
-		X.pixel_y = Pixel_y
+		segment.forceMove(locate(final_x, final_y, segment.z))
+		segment.pixel_x = origin_px + Pixel_x
+		segment.pixel_y = origin_py + Pixel_y
 		CHECK_TICK
 
 /obj/effect/ebeam
 	mouse_opacity = MOUSE_OPACITY_TRANSPARENT
+	layer = ABOVE_ALL_MOB_LAYER
 	anchored = TRUE
+	var/emissive = TRUE
 	var/datum/beam/owner
 
+
+/obj/effect/ebeam/Initialize(mapload, beam_owner)
+	owner = beam_owner
+	return ..()
+
+/obj/effect/ebeam/update_overlays()
+	. = ..()
+	if(!emissive)
+		return
+	var/mutable_appearance/emissive_overlay = emissive_appearance(icon, icon_state, layer, alpha, appearance_flags, filters)
+	emissive_overlay.transform = transform
+	emissive_overlay.alpha = alpha
+	. += emissive_overlay
```

Comment:
You need to add a lum source using the define here, otherwise it won't show in byond darkness